src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -398,31 +398,26 @@ private void AddCredentialToCache(Uri serverUri, CURLAUTH authAvail, NetworkCred
             }
         }
 
-        private void AddResponseCookies(Uri serverUri, HttpResponseMessage response)
+        private void AddResponseCookies(EasyRequest state, string cookieHeader)
         {
             if (!_useCookie)
             {
                 return;
             }
 
-            if (response.Headers.Contains(HttpKnownHeaderNames.SetCookie))
+            Uri serverUri = state._requestMessage.RequestUri;
```

Comment:
(sorry, missed this comment. I did get the email notification but I could not find the comment here till now. )

I see my mistake in the code. In WinHttpHandler, adding into CookieContainer happens [before setting the url](https://github.com/dotnet/corefx/blob/master/src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs#L183)